src/benchmarks/real-world/Microsoft.ML.Benchmarks/Microsoft.ML.Benchmarks/PredictionEngineBenchcs.cs

```diff
@@ -0,0 +1,200 @@
+                .Append(new SdcaMultiClassTrainer(env, "Label", "Features", advancedSettings: (s) => { s.NumThreads = 1; s.ConvergenceTolerance = 1e-2f; }));
+
+            var model = pipeline.Fit(data);
+
+            _irisModel = model.CreatePredictionEngine<IrisData, IrisPrediction>(env);
+        }
+
+        [GlobalSetup(Target = nameof(MakeSentimentPredictions))]
+        public void SetupSentimentPipeline()
+        {
+            _sentimentExample = new SentimentData()
+            {
+                SentimentText = "Not a big fan of this."
+            };
+
+            string _sentimentDataPath = Program.GetInvariantCultureDataPath("wikipedia-detox-250-line-data.tsv");
+
+            var env = new MLContext(seed: 1, conc: 1);
+            var reader = new TextLoader(env, columns: new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.BL, 0),
+                            new TextLoader.Column("SentimentText", DataKind.Text, 1)
+                        },
+                        hasHeader: true
+                    );
+
+            IDataView data = reader.Read(_sentimentDataPath);
+
+            var pipeline = new TextFeaturizingEstimator(env, "SentimentText", "Features")
+                .Append(new SdcaBinaryTrainer(env, "Label", "Features", advancedSettings: (s) => { s.NumThreads = 1; s.ConvergenceTolerance = 1e-2f; }));
+
+            var model = pipeline.Fit(data);
+
+            _sentimentModel = model.CreatePredictionEngine<SentimentData, SentimentPrediction>(env);
+        }
+
+        [GlobalSetup(Target = nameof(MakeBreastCancerPredictions))]
+        public void SetupBreastCancerPipeline()
+        {
+            _breastCancerExample = new BreastCancerData()
+            {
+                Features = new[] { 5f, 1f, 1f, 1f, 2f, 1f, 3f, 1f, 1f }
+            };
+
+            string _breastCancerDataPath = Program.GetInvariantCultureDataPath("breast-cancer.txt");
+
+            var env = new MLContext(seed: 1, conc: 1);
+            var reader = new TextLoader(env, columns: new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.BL, 0),
+                            new TextLoader.Column("Features", DataKind.R4, new[] { new TextLoader.Range(1, 9) })
+                        },
+                        hasHeader: false
+                    );
+
+            IDataView data = reader.Read(_breastCancerDataPath);
+
+            var pipeline = new SdcaBinaryTrainer(env, "Label", "Features", advancedSettings: (s) => { s.NumThreads = 1; s.ConvergenceTolerance = 1e-2f; });
+
+            var model = pipeline.Fit(data);
+
+            _breastCancerModel = model.CreatePredictionEngine<BreastCancerData, BreastCancerPrediction>(env);
+        }
+
+        [Benchmark]
+        public void MakeIrisPredictions()
+        {
+            for (int i = 0; i < 10000; i++)
+            {
+                _irisModel.Predict(_irisExample);
+            }
```

Comment:
sorry, I missed this before. It's not recommended to have your own iterations loop in a benchmark - Benchmark.NET scales the iterations itself so you shouldn't need to do this.

Reply:
@billwert is correct, this is not needed (BDN scales the number of invocations per iteration based on IterationTime setting)